subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -980,11 +981,48 @@ int nrf_cloud_obj_location_request_create(struct nrf_cloud_obj *const obj,
 		goto cleanup;
 	}
 
-	/* By default, nRF Cloud will send the location to the device */
-	if (!request_loc &&
-	    nrf_cloud_obj_num_add(&data_obj, NRF_CLOUD_LOCATION_KEY_DOREPLY, 0, false)) {
+	if (config && (obj->type == NRF_CLOUD_OBJ_TYPE_JSON)) {
+		/* Assume no differences from defaults. */
 		err = -ENOMEM;
-		goto cleanup;
+		if (config->do_reply != true) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_DOREPLY,
+						     config->do_reply, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (config->hi_conf != false) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_HICONF,
+						     config->hi_conf, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (config->fallback != true) {
+			err = nrf_cloud_obj_bool_add(&config_obj,
+						     NRF_CLOUD_LOCATION_JSON_KEY_FALLBACK,
+						     config->fallback, false);
+			if (err) {
+				goto cleanup;
+			}
+		}
+		if (!err) {
+			/* At least one entry differed from defaults, so add it to the obj. */
+			err = nrf_cloud_obj_object_add(obj, NRF_CLOUD_LOCATION_JSON_KEY_CONFIG,
+						       &config_obj, false);
+			if (err) {
+				goto cleanup;
+			}
+			/* The config object now belongs to the location request object */
+			config_obj.json = NULL;
+		} else {
+			(void)nrf_cloud_obj_free(&config_obj);
```

Comment:
this else can be removed (assuming you init the config_obj inside the if statement).

Reply:
Was this free needed? I guess it's needed when there are no errors up to this point. Another free call if there is an error later in the function will occur but I guess that doesn't matter...

Reply:
I set `err = -ENOMEM` before checking the 3 flags to see if they differ from the default (I basically use `err` for two purposes).

If any of them differ, I add JSON bools for them and set the values accordingly.  The act of adding sets the `err` variable to 0 when successful; when non-zero, we bail out to `cleanup`.

If none of them differ, `err == -ENOMEM` still, which indicates that `config_obj` was not needed after all, so I delete it.

This works fine but is obscure. I'll make it clearer.

Reply:
> Another free call if there is an error later in the function will occur but I guess that doesn't matter...

Is double free an issue? It can happen when we go to cleanup from line 1041 or 1047.

Looking at `nrf_cloud_obj_free()`, I guess you can call it twice for the same object and the underlying JSON etc. data is only freed on the first instance. The second call will simply return `-ENOTSUP`, right?